### PR TITLE
Add option to grow task stack upwards, address #1

### PR DESCRIPTION
**Pull request by j-hui**
@tjdevries something like this?

I'm considering whether this makes sense as the default behavior. On one hand, being less jumpy is always a good thing. On the other hand, I'm slightly not used to how this looks.

### CODE REVIEW

File: lua/fidget.lua [hunk 1]
```diff
@@ -63,7 +64,11 @@ function base_fidget:fmt()
   self.max_line_len = #line
   for _, task in pairs(self.tasks) do
     line = options.fmt.task(task.title, task.message, task.percentage)
-    table.insert(self.lines, line)
+    if options.fmt.stack_upwards then
```
Comment (j-hui) [inline lua/fidget.lua:67]: Checking this in a loop every time `:fmt()` is called seems expensive, but this is also the kind of thing a decent optimizer will pick up immediately. Does Neovim's embedded Lua interpreter optimize this?

I suppose lifting this into a local variable is not that much less readable.

Comment (tjdevries) [reply lua/fidget.lua:67]: you should consider checking a variable free.

Try checking a variable in a loop 1 million times and tell me how long it takes :)

**Comment by tjdevries** [comment]
Don't mind either way for default -- I like how little it jumps around and the "bright" things stay still. I can test if you want -- looks good from casual review.

File: lua/fidget.lua (hunk 1 above)
Comment (j-hui) [reply lua/fidget.lua:67]: Awesome. I'm still new to Lua so I'm not famliar with where the usual bottlenecks are, or when the optimizer kicks in.

**Comment by j-hui** [comment]
I'll make it the default behavior. Thanks for the review and comments @tjdevries !

**Pull request merged**